File: model/modules/EVI/EVI_forcing/EVI_forcing.m
function [f,fe,fx,s,d,p]=EVI_forcing(f,fe,fx,s,d,p,info,tix)
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
% sets the value of s.cd.EVI from the forcing in every time step
%
% Inputs:
%   - f.EVI read from the forcing data set
%
% Outputs:
%   - s.cd.EVI: the value of EVI for current time step
%
% Modifies:
%   - s.cd.EVI
%
% References:
%   - 
%
% Created by:
%   - Sujan Koirala (skoirala)
%
% Versions:
%   - 1.0 on 11.11.2019 (skoirala): 
%
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

%% 
s.cd.EVI = f.EVI(:,tix);

end

File: src/models/EVI/EVI_forcing.jl
export EVI_forcing, EVI_forcing_h
"""
sets the value of land.states.EVI from the forcing in every time step

# Parameters:
$(PARAMFIELDS)
"""
@bounds @describe @units @with_kw struct EVI_forcing{T} <: EVI
	noParameter::T = nothing | nothing | nothing | nothing
end

function precompute(o::EVI_forcing, forcing, land, infotem)
	# @unpack_EVI_forcing o
	return land
end

function compute(o::EVI_forcing, forcing, land, infotem)
	@unpack_EVI_forcing o

	## unpack variables
	@unpack_land begin
		EVI ∈ forcing
	end

	## pack variables
	@pack_land begin
		EVI ∋ land.states
	end
	return land
end

function update(o::EVI_forcing, forcing, land, infotem)
	# @unpack_EVI_forcing o
	return land
end

"""
sets the value of land.states.EVI from the forcing in every time step

# precompute:
precompute/instantiate time-invariant variables for EVI_forcing

# compute:
Enhanced vegetation index using EVI_forcing

*Inputs:*
 - forcing.EVI read from the forcing data set

*Outputs:*
 - land.states.EVI: the value of EVI for current time step

# update
update pools and states in EVI_forcing
 - land.states.EVI

# Extended help

*References:*
 -

*Versions:*
 - 1.0 on 11.11.2019 [skoirala]:  

*Created by:*
 - Sujan Koirala [skoirala]
"""
function EVI_forcing_h end